### PR TITLE
Added latest versions of SQLAlchemy 1.3.X, 1.2.X and 1.1.X

### PR DESCRIPTION
**Pull request by wspear**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-sqlalchemy/package.py [hunk 1]
```diff
@@ -10,6 +10,9 @@ class PySqlalchemy(PythonPackage):
     """The Python SQL Toolkit and Object Relational Mapper"""
 
     homepage = 'http://www.sqlalchemy.org/'
-    url      = "https://pypi.io/packages/source/S/SQLAlchemy/SQLAlchemy-1.0.12.tar.gz"
+    url      = "https://pypi.io/packages/source/S/SQLAlchemy/SQLAlchemy-1.3.9.tar.gz"
 
+    version('1.3.9', sha256='272a835758908412e75e87f75dd0179a51422715c125ce42109632910526b1fd')
+    version('1.2.19', sha256='5bb2c4fc2bcc3447ad45716c66581eab982c007dcf925482498d8733f86f17c7')
+    version('1.1.18', sha256='8b0ec71af9291191ba83a91c03d157b19ab3e7119e27da97932a4773a3f664a9')
     version('1.0.12', '6d19ef29883bbebdcac6613cf391cac4')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sqlalchemy/package.py]: Missing dependencies:
```python
depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
depends_on('py-setuptools', type='build')
```
There's also some test dependencies and extras you can add if you want.

Comment (wspear) [reply var/spack/repos/builtin/packages/py-sqlalchemy/package.py]: I added the test dependencies as well but running the tests fails like this:

spack$ spack install --test=all py-sqlalchemy^python@3.7.4
==> Error: maximum recursion depth exceeded

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sqlalchemy/package.py]: Huh, I guess there's a circular dependency somewhere. Does `spack install --test=root` work though?

Comment (wspear) [reply var/spack/repos/builtin/packages/py-sqlalchemy/package.py]: --test=root runs normally but the test itself fails. I'm guessing an unexpected version/interface-change based on the error.

` goultlphqgckjwegkux/spack-src/lib/sqlalchemy/cutils.cpython-37m-x86_64-linux-gnu.so
     259    ERROR: usage: setup.py [options] [file_or_dir] [file_or_dir] [...]
  >> 260    setup.py: error: unrecognized arguments: --max-worker-restart=5
     261      inifile: /tmp/wspear/spack-stage/py-sqlalchemy-1.3.9-6cnjrpk5ujig5goultlphqgckjwegkux/spack-src/setup.cfg
     262      rootdir: /tmp/wspear/spack-stage/py-sqlalchemy-1.3.9-6cnjrpk5ujig5goultlphqgckjwegkux/spack-src
     263    
`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sqlalchemy/package.py]: Yeah, a lot of Python packages have unit tests, but either they aren't uploaded in the PyPI tarball, or they aren't properly setup to be run with `python setup.py test`. I think it's fine to keep the dependencies and let someone else sort out the mess if they really care.

**Pull request merged**